### PR TITLE
Fix deployer teardown test logic

### PR DESCRIPTION
**Pull request by cds-amal**
This PR aims to remove mocha's `--exit` flag from the deployer tests. We shouldn't be using this flag because it hides potential issues with Ganache and Truffle's interaction.

Currently, there is a blocker, an issue with how Ganache.Provider eager mode and `provider.disconnect()` works, or Truffle's usage of it! See: https://github.com/trufflesuite/ganache/issues/3293

### CODE REVIEW

File: packages/deployer/package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
   "main": "index.js",
   "scripts": {
     "prepare": "exit 0",
-    "test": "mocha --no-warnings --timeout 10000 --exit"
+    "test": "mocha --no-warnings --timeout 10000",
+    "testx": "why-is-node-running $(yarn bin)/mocha --no-warnings --timeout 10000"
```
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/deployer/package.json]: oh yeah as the root seems to make sense. I mean... I wouldn't mind having this be convenient :)

**Comment by cds-amal** [comment]
Note: this may be unblocked! See: https://github.com/trufflesuite/ganache/pull/3433

**Comment by cds-amal** [comment]
When using `instamine eager mode` we seem to be experiencing the issue described in [Ganache#2083](https://github.com/trufflesuite/ganache/issues/2083) causing the test process to hang. This seems like a legitimate use of `mocha --exit`. 

TODO: Will still have to write up the sequence of events that causes the process to hang.

**Comment by cds-amal** [comment]
Closing this in favor of #5832

**Pull request closed without merging**